### PR TITLE
Added notes to person view

### PR DESCRIPTION
**Pull request by hortha**
### Context

Users need a way to view existing notes attached to a Person record in the TRS Console.

### Changes proposed in this pull request

Using the timeline component [Timeline - MOJ Pattern Library](https://design-patterns.service.justice.gov.uk/components/timeline/) experiment with adding a Notes tab to the Person view that displays the same notes available in the CRM UI view.

### Guidance to review

CRM query changes + UI + tests (including helpers for creating notes and tasks)

### Checklist

-   [x] Attach to Trello card
-   [x] Rebased master
-   [ ] Cleaned commit history
-   [x] Tested by running locally

### CODE REVIEW

**Comment by hortha** [comment]
[Spike adding notes to person view](https://trello.com/c/li4GtCBF/1258-spike-adding-notes-to-person-view)

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/ChangeLog.cshtml [hunk 1]
```diff
@@ -11,4 +11,40 @@
...
 }
 
-<p class="govuk-body">Change log goes here</p>
+@if (Model.TimelineItems is null || Model.TimelineItems.Length == 0)
+{
+    <p class="govuk-body" data-testid="no-changes">No changes</p>
+}
+else
+{
+    var itemNumber = 0;
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/ChangeLog.cshtml:20]: Is this used?

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/ChangeLog.cshtml [hunk 2]
```diff
@@ -11,4 +11,40 @@
     ViewBag.Title = Model.Name;
 }
 
-<p class="govuk-body">Change log goes here</p>
+@if (Model.TimelineItems is null || Model.TimelineItems.Length == 0)
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/ChangeLog.cshtml:14]:
```suggestion
@if (Model.TimelineItems!.Length == 0)
```

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/ChangeLog.cshtml (hunk 1 above)
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/ChangeLog.cshtml:20]: not any more!

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/ChangeLog.cshtml [hunk 3]
```diff
@@ -11,4 +11,40 @@
...
+                <div class="moj-timeline__header">
+                    <h2 class="moj-timeline__title" data-testid="timeline-item-title">@timelineItem.Title</h2>
+                    <p class="moj-timeline__byline" data-testid="timeline-item-user">by @timelineItem.User</p>
+                    @if (timelineItem.Status is not null)
+                    {
+                        var tagClass = timelineItem.Status switch
+                        {
+                            ChangeLogModel.TimelineItemStatus.Overdue => "govuk-tag--red",
+                            ChangeLogModel.TimelineItemStatus.Closed => "govuk-tag--grey",
+                            _ => "govuk-tag"
```
Comment (gunndabad) [inline TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/ChangeLog.cshtml:35]: I think the tag helper will add `govuk-tag` already
```suggestion
                            _ => ""
```

**Review by gunndabad — APPROVED** [review]

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/ChangeLog.cshtml (hunk 1 above)
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/ChangeLog.cshtml]: Done

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/ChangeLog.cshtml (hunk 2 above)
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/ChangeLog.cshtml]: Done

File: TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/ChangeLog.cshtml (hunk 3 above)
Comment (hortha) [reply TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/Persons/PersonDetail/ChangeLog.cshtml]: Done

File: TeachingRecordSystem/tests/TeachingRecordSystem.TestCommon/CrmTestData.CreateCrmTask.cs [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+        private Guid? _personId = null;
+        private string? _category = null;
+        private string? _subject = null;
+        private string? _description = null;
+        private DateTime? _dueDate = null;
+        private TaskState? _stateCode = null;
+
+        public CreateCrmTaskBuilder WithPersonId(Guid personId)
+        {
+            if (_personId is not null)
```
[the remark this answers is not in the document]
Comment (hortha) [reply TeachingRecordSystem/tests/TeachingRecordSystem.TestCommon/CrmTestData.CreateCrmTask.cs]: Done

**Pull request merged**